core/math/geometry_2d.h

```diff
@@ -249,6 +253,28 @@ class Geometry2D {
 		return -1;
 	}
 
+	static bool segment_intersects_rect(const Vector2 &p_from, const Vector2 &p_to, const Rect2 &p_rect) {
+		if (p_rect.has_point(p_from) || p_rect.has_point(p_to)) {
+			return true;
+		}
+
+		const Vector2 rect_points[4] = {
+			p_rect.position,
+			p_rect.position + Vector2(p_rect.size.x, 0),
+			p_rect.position + p_rect.size,
+			p_rect.position + Vector2(0, p_rect.size.y)
```

Comment:
You could alternatively use `position`/`end` and their individual `x` and `y`, but I'm not sure if it would help readability. This is probably better.

Reply:
I think I'll leave it this way.